### PR TITLE
gh-93939: Create and install scripts in Makefile (GH-94324)

### PR DESCRIPTION
**Pull request by tiran**
Scripts for 2to3, idle, and pydoc are now created and installed by make.

* Issue: gh-93939

### CODE REVIEW

File: Makefile.pre.in [hunk 1]
```diff
@@ -2121,6 +2123,28 @@ python-config: $(srcdir)/Misc/python-config.in Misc/python-config.sh
...
+BUILD_SCRIPTS_DIR=build/scripts-$(VERSION)
+SCRIPT_2TO3=$(BUILD_SCRIPTS_DIR)/2to3-$(VERSION)
+SCRIPT_IDLE=$(BUILD_SCRIPTS_DIR)/idle$(VERSION)
+SCRIPT_PYDOC=$(BUILD_SCRIPTS_DIR)/pydoc$(VERSION)
+
+$(BUILD_SCRIPTS_DIR):
+	$(MKDIR_P) $@
+
+$(SCRIPT_2TO3): $(srcdir)/Tools/scripts/2to3 $(BUILD_SCRIPTS_DIR)
+	$(MKDIR_P) $(BUILD_SCRIPTS_DIR)
```
Comment (arhadthedev) [inline Makefile.pre.in]: Isn't the directory creation covered by dependency (in line 2134) on rule `$(BUILD_SCRIPTS_DIR)`?

Comment (tiran) [reply Makefile.pre.in]: You have not read the commit message ;)

Comment (arhadthedev) [reply Makefile.pre.in]: > debug mac builds

My bad; I had a habit to scan through commits in case of incomprehensible diff on *File changed* tab.

**Comment by vstinner** [comment]
I'm not sure if it's related, but all tests fail on parsing JSON on the "wasm32-emscripten node (pthreads) 3.x" buildbot since this change: https://buildbot.python.org/all/#/builders/1050/builds/36

**Pull request merged**